web_external/models/UserModel.js

```diff
@@ -33,39 +34,53 @@ UserModel.prototype.canAcceptTerms = function () {
     return this.get('permissions').acceptTerms === true;
 };
 
-UserModel.prototype.setAcceptTerms = function (successCallback) {
+UserModel.prototype.setAcceptTerms = function () {
+    const deferred = $.Deferred();
     restRequest({
         path: 'user/acceptTerms',
         type: 'POST'
-    }).done((resp) => {
+    })
+    .done((resp) => {
         if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
             // Directly update user permissions
             this.get('permissions').acceptTerms = true;
             this.trigger('change:permissions');
-            successCallback(resp);
+            deferred.resolve(resp);
+        } else {
+            // This should not fail
+            deferred.reject(resp);
         }
-        // This should not fail
+    })
+    .fail((resp) => {
+        deferred.reject(resp);
     });
+    return deferred.promise();
```

Comment:
I mostly agree. We'll still have to create a new Deferred here, but we can do it within the `.then`. Ideally, we should just be able to use `return` and `throw` within `.then`, but jQuery 2 doesn't support that yet.